### ISSUE TITLE
Update/Add Windows SDK and WDK (10.0.26100)

### ISSUE THREAD

**Issue by jxy-s** [actions/runner-images#10349]
### Tool name

Windows SDK and WDK 10.0.26100

### Tool license

MICROSOFT SOFTWARE LICENSE

### Add or update?

- [X] Add
- [X] Update

### Desired version

10.0.26100

### Brief description of tool

https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/
> The Windows SDK (10.0.26100) for Windows 11 provides the latest headers, libraries, metadata, and tools for building Windows applications. Use this SDK to build Universal Windows Platform (UWP) and Win32 applications for Windows 11, version 24H2 preview and previous Windows releases.

https://learn.microsoft.com/en-us/windows-hardware/drivers/download-the-wdk
> The WDK is used to develop, test, and deploy drivers for Windows. The most recent public release is WDK 10.0.26100.1 (released May 22, 2024).

### URL for tool's homepage

https://learn.microsoft.com/en-us/windows-hardware/drivers/download-the-wdk

### Provide a basic test case to validate the tool's functionality.

```bash
There are libraries shipped with the latest WDK that are not available and as such many builds are failing, see:
https://github.com/winsiderss/systeminformer/actions/runs/9650107442/job/26615057348

I have had to temporarily disable our CI:
https://github.com/winsiderss/systeminformer/commit/4666308dc61a85e1ce733deb9107f752a3b90aed

The following issue is related to this:
https://github.com/winsiderss/systeminformer/issues/2109
```

### Platforms where you need the tool

- [X] Azure DevOps
- [X] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [X] Windows Server 2019
- [X] Windows Server 2022

### Can this tool be installed during the build?

```bash
No, not really, these are build tools that many build tools depend on to begin with. Could hack something together but really these kits should be included with the images.
```

**Comment by jxy-s** [comment]
Apologies for creating a duplicate, a previous issue was open but was closed: https://github.com/actions/runner-images/issues/10012

Since this still needs to be taken care of, I've created this new issue.

This is blocking the following: https://github.com/winsiderss/systeminformer/issues/2109

**Comment by RaviAkshintala** [comment]
@jxy-s  Thank you for bringing this issue to us. We are looking into this issue and will update you on this issue after investigating.

**Comment by middlemose** [comment]
@jxy-s Please take a look at our repo [Windows-driver-samples](https://github.com/microsoft/Windows-driver-samples) and review our pipeline as well as how we build our samples.

Please consider using [NuGet](https://www.nuget.org/packages/Microsoft.Windows.WDK.x64) as recommended approach to ingest the latest WDK. Direct link to documentation (using NuGet) to build locally [Option A: Use WDK NuGet Packages](https://github.com/microsoft/Windows-driver-samples/blob/main/Building-Locally.md#option-a-use-wdk-nuget-packages:~:text=Option%20A%3A%20Use%20WDK%20NuGet%20Packages)

FYI @NeoAdonis and @prashantchahar

**Comment by jxy-s** [comment]
@middlemose thanks for the pointers on the WDK. I'll take a look. That should unblock me for the time being.

I still believe the images should be updated to include the latest build tools (both SDK and WDK).

**Comment by zufuliu** [comment]
See https://github.com/zufuliu/notepad4/issues/839, the SDK does not install libs for 32-bit ARM.

**Comment by ankushg** [comment]
Have a PR up to add Windows SDK 26100 to the windows-latest image: https://github.com/actions/runner-images/pull/10452

Wasn't sure about 
- Windows Driver Kit because it looks like you can't install multiple versions simultaneously
- Adding it to the Windows Server 2019 image

**Comment by RaviAkshintala** [comment]
@jxy-s @ankushg   Windows Driver Kit `10.0.26100` is available now with the latest release (`20240825.1`) . Closing this issue, Thank you.

**Comment by ankushg** [comment]
hi @RaviAkshintala this is actually not fully closed - Windows Driver Kit and Windows SDK follow the same versioning, but are distributed separately

It looks like the latest release **does** include Windows Driver Kit 10.0.26100, but **does not** include the corresponding Windows SDK

See my PR here: https://github.com/actions/runner-images/pull/10452/files

**Comment by middlemose** [comment]
FYI @RaviAkshintala Latest Windows Driver Kit is still not installed. 

I tested latest runner image and this is the output.
![image](https://github.com/user-attachments/assets/acb61ac4-92c8-43fe-9fba-6709a05d424b)

However, latest extension is installed (which is nice to see as proper component install :-} ). 

WDK team still recommends to not install the WDK and help promote use of NuGet as the recommended way to use WDK. When using the NuGet the matching SDK NuGet dependency comes with it allowing developers to never be concerned with SDK pairing.
